storage/replica.go

```diff
@@ -2032,7 +2060,19 @@ var enableQuiescence = envutil.EnvOrDefaultBool("COCKROACH_ENABLE_QUIESCENCE", f
 // follower which will send a MsgProp to the leader that will unquiesce the
 // leader. If the leader of a quiesced range dies, followers will not notice,
 // though any request directed to the range will eventually end up on a
-// follower which will unquiesce the follower and lead to an election.
+// follower which will unquiesce the follower and lead to an election. When a
+// follower unquiesces for a reason other than receiving a raft message or
+// proposing a raft command, it wakes the leader by sending an empty message
+// proposal. This acts a failsafe against bugs in which a follower is left
+// unquiesced while the leader is quiesced.
+//
+// TODO(peter): There remains a scenario in which a follower is left unquiesced
+// while the leader is quiesced: the follower's receive queue is full and the
```

Comment:
The system will recover from either the quiesce or empty `MsgProp` being dropped. We'll get an unnecessary election and worst case we'll see the behavior of #9372 and a few second hiccup to the range. If we fix the election machinery so the spurious election doesn't fail (simulated by disabling `CheckQuorum`) the hiccup becomes microscopic.